src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -1637,6 +1800,19 @@ int SketchObject::fillet(int GeoId1, int GeoId2,
         GeoId2 < 0 || GeoId2 > getHighestCurveIndex())
         return -1;
 
+    // If either of the two input lines are locked, don't try to trim since it won't work anyway
+    if (trim) {
+        const std::vector<Constraint *> &constraints = this->Constraints.getValues();
+        for (int i=0; i< int(constraints.size()); i++) {
+            Constraint *c = constraints[i];

```

Comment:
Same as before: better with a range-based `for`:
```suggestion
        for (auto c :  this->Constraints.getValues()) {
```